From 8ce19bb2ca59eb8fb782971239f474854547873e Mon Sep 17 00:00:00 2001
From: Dirk-Jan Faber <dfaber@victronenergy.com>
Date: Mon, 3 Aug 2026 12:32:12 +0200
Subject: [PATCH 3/3] package.json: add socketcan package

Venus OS no longer installes packages listed as optional [1]. This also applies
to dependencies in the tree.

canboatjs[2], the package used by signalk for N2K communication, has socketcan
defined as an optional, hence add it here as needed.

[1] https://github.com/victronenergy/venus/issues/1304
[2] https://github.com/canboat/canboatjs
---
 package.json | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/package.json b/package.json
index 96d71b9a..afa5fe08 100644
--- a/package.json
+++ b/package.json
@@ -147,7 +147,8 @@
     "serialport": "^11.0.1",
     "signalk-n2kais-to-nmea0183": "^2.0.3",
     "signalk-to-nmea2000": "^2.24.0",
-    "signalk-venus-plugin": "^2.4.1"
+    "signalk-venus-plugin": "^2.4.1",
+    "socketcan": "^4.2.4"
   },
   "devDependencies": {
     "@eslint-react/eslint-plugin": "^1.53.1",
-- 
2.43.0

